### PR TITLE
PA-2303, PA-2304, PA-2305, PA-2307, PA-2306, PA-2308

### PR DESCRIPTION
**Pull request by devinleighsmith**
Updates based on QA review.

### CODE REVIEW

File: frontend/src/components/common/form/ParentSelect.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import { getIn, useFormikContext, setIn } from 'formik';
+import { getIn, useFormikContext } from 'formik';
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/ParentSelect.tsx:1]: These changes fix the ParentSelect component such that it no longer sets agencyId to an object.

File: frontend/src/components/common/form/StepForm/context.tsx [hunk 2]
```diff
@@ -119,10 +119,11 @@ export const StepperFormProvider: React.FC<IStepperFormProviderProps> = ({
   };
 
   const gotoStep = (index: number) => {
-    if (!validateCurrentStep()) {
-      return false;
-    }
     if (index >= 0 && index < steps.length) {
+      const currentIndex = getIn(values, `tabs.${values.activeTab}.activeStep`);
+      if (!validateCurrentStep() && index > currentIndex) {
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/StepForm/context.tsx:124]: add an override for PA-2307 such that as long as the user is going to a previous step ignore validation failures.

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx [hunk 3]
```diff
@@ -56,7 +58,7 @@ export const saveParcelDataLayerResponse = (
       }),
     );
   } else {
-    toast.warning(`Failed to find PID/PIN. Ensure that the searched PID/PIN is valid`);
+    toast.warning(`Failed to find parcel layer data. Ensure that the search criteria is valid`);
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:61]: This is used for pid/pin and also lat/lng searches.

File: frontend/src/utils/YupSchema.ts [hunk 4]
```diff
@@ -216,6 +216,9 @@ export const ParcelSchema = Yup.object()
       financials: Yup.array()
         .compact((financial: any) => financial.year !== currentYear)
         .of(FinancialYear),
+      agencyId: Yup.number()
```
Comment (devinleighsmith) [inline frontend/src/utils/YupSchema.ts:219]: the agency field is actually required (missing in design).

File: frontend/src/features/properties/hooks/useParcelLayerData.tsx [hunk 5]
```diff
@@ -66,7 +66,7 @@ const setParcelFieldsFromLayerData = (
     }
 
     const layerParcelData = layerData!.data;
-    const pid = layerData?.data.PID || layerData?.data.PID_NUMBER.toString();
+    const pid = layerData?.data?.PID || layerData?.data?.PID_NUMBER?.toString();
```
Comment (devinleighsmith) [inline frontend/src/features/properties/hooks/useParcelLayerData.tsx:69]: fix behaviour if user clicks on an empty part of the map.

File: frontend/src/features/properties/filter/PropertyFilter.tsx [hunk 6]
```diff
@@ -99,7 +99,7 @@ export const PropertyFilter: React.FC<IPropertyFilterProps> = ({
   };
 
   const changeFilter = (values: IPropertyFilter) => {
-    const agencyIds = (values.agencies as any)?.value ?? '';
+    const agencyIds = values.agencies;
```
Comment (devinleighsmith) [inline frontend/src/features/properties/filter/PropertyFilter.tsx:102]: don't need to do the .value with the changes to ParentSelect.

File: frontend/src/features/mapSideBar/hooks/useQueryParamSideBar.tsx [hunk 7]
```diff
@@ -113,15 +112,6 @@ const useQueryParamSideBar = (): IMapSideBar => {
     setShowSideBar: setShow,
     size: sideBarSize,
     parcelId,
-    overrideParcelId: parcelId => {
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/hooks/useQueryParamSideBar.tsx:116]: remove dead code.

File: frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx [hunk 8]
```diff
@@ -188,6 +188,8 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
               handleParcelDataLayerResponse(response, dispatch);
             }
           });
+        } else {
+          toast.warning('Unable to find any details for the clicked location.');
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx:192]: display a message when there is no result when a marker is dropped on the map.

File: frontend/src/features/mapSideBar/SidebarContents/subforms/ParcelIdentificationForm.tsx [hunk 9]
```diff
@@ -245,6 +246,7 @@ export const ParcelIdentificationForm: React.FC<IIdentificationProps> = ({
           <Form.Row>
             <Form.Label>{agency?.parent ? 'Sub Agency' : 'Agency'}</Form.Label>
             <ParentSelect
+              required
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/subforms/ParcelIdentificationForm.tsx:249]: agency is a required field.

File: frontend/src/features/mapSideBar/SidebarContents/subforms/ParcelIdentificationForm.tsx [hunk 10]
```diff
@@ -175,6 +175,7 @@ export const ParcelIdentificationForm: React.FC<IIdentificationProps> = ({
             displayErrorTooltips
           />
           <SearchButton
+            disabled={!geocoderResponse}
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/subforms/ParcelIdentificationForm.tsx:178]: Prompt the user to enter an address properly before searching.

File: frontend/src/features/mapSideBar/SidebarContents/subforms/LandReviewPage.tsx [hunk 11]
```diff
@@ -115,7 +116,7 @@ export const LandReviewPage: React.FC<any> = (props: IReviewProps) => {
                 <TextArea
                   required={true}
                   disabled={editInfo.identification}
-                  field={withNameSpace('description')}
+                  field={withNameSpace('landLegalDescription')}
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/subforms/LandReviewPage.tsx:119]: duplicated field.

File: frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx [hunk 12]
```diff
@@ -372,14 +372,14 @@ const LandForm: React.FC<IParentLandForm> = (props: IParentLandForm) => {
       route: 'usage',
       title: 'Usage',
       completed: false,
-      canGoToStep: !!props.disabled,
+      canGoToStep: !!initialValues?.data?.id,
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx:375]: if this is a view/update allow direct step navigation.

Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx:375]: PA-2303

File: frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx [hunk 13]
```diff
@@ -389,16 +389,16 @@ const LandForm: React.FC<IParentLandForm> = (props: IParentLandForm) => {
   if (!!props.initialValues?.buildings?.length) {
     steps.push({
       route: 'associatedLand',
-      title: 'View Associated Land',
+      title: 'View Buildings',
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx:392]: PA-2304

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 14]
```diff
@@ -160,16 +160,20 @@ const Map: React.FC<MapProps> = ({
   });
 
   const lastZoom = useSelector<RootState, number>(state => state.mapViewZoom) ?? zoomProp;
+  const [zoom, setZoom] = useState(lastZoom);
   useEffect(() => {
-    dispatch(setMapViewZoom(smallScreen ? 4.9 : 5.5));
-  }, [dispatch, smallScreen]);
+    if (lastZoom === DEFAULT_MAP_ZOOM) {
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/Map.tsx:165]: PA-2305 actually update lastzoom when the zoom changes, previously lastzoom is only set to a default value and never changed again. This is what was causing the map to zoom out.

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx [hunk 15]
```diff
@@ -84,17 +86,42 @@ export const handleParcelDataLayerResponse = (
  * @param geometry the name of the geometry in the feature collection
  */
 export const useLayerQuery = (url: string, geometryName: string = 'SHAPE'): IUserLayerQuery => {
+  const instance = axios.create();
+  const dispatch = useDispatch();
+
+  instance.interceptors.request.use(
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:92]: PA-2306 increase user feedback when loading search data by displaying loading bar.

File: frontend/src/features/mapSideBar/SidebarContents/AssociatedLandForm.tsx [hunk 16]
```diff
@@ -553,10 +552,6 @@ const AssociatedLandForm: React.FC<IAssociatedLandParentForm> = (
             initialValues={props.initialValues}
             isAdmin={props.isAdmin}
           />
-          <Prompt
```
Comment (devinleighsmith) [inline frontend/src/features/mapSideBar/SidebarContents/AssociatedLandForm.tsx:556]: Unfortunately this has too many consequences due to the way we are using the url bar to drive behaviour.

Comment (devinleighsmith) [reply frontend/src/features/mapSideBar/SidebarContents/AssociatedLandForm.tsx:556]: PA-2308

**Review by Fosol — APPROVED** [review]

File: frontend/src/actions/parcelsActions.ts [hunk 17]
```diff
@@ -27,15 +28,15 @@ export interface IStoreParcelsAction {
 
 export interface IStoreDraftParcelsAction {
   type: typeof ActionTypes.STORE_DRAFT_PARCEL_RESULTS;
-  draftParcelList: IProperty[];
+  draftParcelList: PointFeature[];
```
Comment (agalbrai) [inline frontend/src/actions/parcelsActions.ts:31]: why this change?

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx [hunk 18]
```diff
@@ -122,7 +127,7 @@ export const InventoryLayer: React.FC<InventoryLayerProps> = ({
 
   return (
     <PointClusterer
-      points={features}
+      points={[...features, ...draftProperties]}
```
Comment (agalbrai) [inline frontend/src/components/maps/leaflet/InventoryLayer.tsx:130]: does this allow there to be a bunch of draft pins grouped together if someone is creating multiple buildings on a parcel?

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx (hunk 3 above)
Comment (agalbrai) [reply frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:61]: I noticed this when searching for an address but the PID/PIN warning popped up, thanks for clarifying!

File: frontend/src/actions/parcelsActions.ts (hunk 17 above)
Comment (devinleighsmith) [reply frontend/src/actions/parcelsActions.ts:31]: With the switch to Geocoder formatted properties in the front-end, draft markers no longer worked because the were not in the correct format. This change saves draft properties in a format that is the same as the rest of our geo-formatted properties.

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx (hunk 18 above)
Comment (devinleighsmith) [reply frontend/src/components/maps/leaflet/InventoryLayer.tsx:130]: this will append all of the draft properties to the list of regular properties.

File: frontend/src/actions/parcelsActions.ts (hunk 17 above)
Comment (agalbrai) [reply frontend/src/actions/parcelsActions.ts:31]: so the draft markers should be appearing properly again?

Comment (devinleighsmith) [reply frontend/src/actions/parcelsActions.ts:31]: once this gets merged, yes.

File: frontend/src/components/common/form/ParentSelect.tsx (hunk 1 above)
Comment (jdtoombs) [reply frontend/src/components/common/form/ParentSelect.tsx:1]: Woo, thank you!

**Review by jdtoombs — APPROVED** [review]
